src/processes/nix_list.rs

```diff
@@ -42,12 +47,16 @@ pub fn active_executables() -> Result<ProcessList> {
                     let executable = e.key().clone();
                     // .file_name() returns `None` if the path terminates in `..`
                     // We use the absolute path in such a case.
-                    let display_name = match path.file_name() {
-                        Some(s) => s.to_string_lossy().to_string(),
-                        None => path.to_string_lossy().to_string(),
+                    let display_name = path
+                        .file_name()
+                        .unwrap_or(path.as_os_str())
+                        .to_string_lossy()
+                        .to_string();
+                    let is_system = match process.effective_user_id() {
+                        Some(id) => id.to_string() == "0",
+                        None => false,
                     };
                     let is_visible = visible.contains(&pid);
-                    let is_system = executable.starts_with("/System/");
```

Comment:
This is a slightly different check now. On my system I see quite a few processes in `/System` running under my uid. Maybe check for either on macOS?

Reply:
> On my system I see quite a few processes in `/System` running under my uid. 

By this do you mean even the `effective_user_id` is your uid for those processes?

Reply:
Good question - I only see uid, macOS `ps` doesn't have a  way to display `euid`. Any idea how I can easily find that out?

Reply:
I think I'm understanding things incorrectly: does `is_system` represent whether a process is started by the system or if the process is started with `sudo` privileges or does it mean something else entirely?

Reply:
Great question, let's add a docstring in the struct if there isn't one already:

The point of `is_system` is to hide (or sort to the bottom) processes in the process selector UI because they are some OS internal things that the user likely is not interested in intercepting.

Does that make sense?

Reply:
I understand what we want to accomplish, but will have to figure out how to differentiate these "system" processes on linux

Reply:
I think euid is a decent heuristic to get started on Linux. We can improve it later. :)

Reply:
what do you think of filtering executables by checking if their path starts with the following to differentiate system processes: `/usr/sbin`, `/sbin`, `/usr/libexec`, `/usr/lib/systemd`?

Reply:
Just from looking at it this sounds super reasonable :)

Reply:
That's not a good heuristic, and it will be getting worse over time. "system service" is an unclear concept, and more and more stuff is moving from `/usr/sbin/` to `/usr/bin`.

See also https://fedoraproject.org/wiki/Features/UsrMove (which moved `/sbin` to `/usr/sbin`, among other things) and https://fedoraproject.org/wiki/Changes/Unify_bin_and_sbin which will make `/usr/sbin` just a symlink to `/usr/bin` in a future Fedora release.

Checking if the user id is <1000 would be a much better heuristic for "process started by the system / a service" IMO.

Reply:
Thanks for pointing it out. Agreed that the current solution isn't correct, but wouldn't checking for uid also mark programs started by the user using `sudo` to be marked as "system" processes.?(or is that what we want?)

Reply:
Isn't that what the difference between "effective" and "real" uid is for?

Reply:
Oh yes, sorry my bad - will make the changes by tomorrow

Reply:
Hm, on second thought, this doesn't seem to work. But checking for uid < 1000 is still much better than checking for sbin, IMO.

Reply:
> this doesn't seem to work

Yes, `.user_id()` seems to return 0 even for programs started using `sudo`, but I've gone ahead implementing it anyways